other/get-debug-information/generate-policy.yaml

```diff
@@ -0,0 +1,81 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: get-debug-data-policy
+  annotations:
+    policies.kyverno.io/title: Collect debug information for pods in crashloopback
+    policies.kyverno.io/category: Other
+    policies.kyverno.io/severity: medium
+    policies.kyverno.io/subject: Deployment
+    kyverno.io/kyverno-version: 1.11.5
+    kyverno.io/kubernetes-version: "1.27"
+    policies.kyverno.io/description: "This policy generates a job which gathers troubleshooting data (including logs, kubectl describe output and events from the namespace) from pods that are in crashloopback and have 3 restarts. This data can further be used to automatically create a jira issue using some kind of automation or another Kyverno policy. For more information on how the image used in this policy was built, see https://github.com/nirmata/SRE-Operational-Usecases/tree/main/get-troubleshooting-data/get-debug-data. "
+spec:
+  rules:
+  - name: get-debug-data-policy-rule
+    match:
+      any:
+      - resources:
+          kinds:
+          - v1/Pod.status
+    context:
+    - name: pdcount
+      apiCall:
+        urlPath: "/api/v1/namespaces/{{request.namespace}}/pods?labelSelector=requestpdname=pod-{{request.object.metadata.name}}"
+        jmesPath: "items | length(@)"
+    preconditions:
+      all:
+      - key: "{{ sum(request.object.status.containerStatuses[*].restartCount || [`0`]) }}"
+        operator: Equals
+        value: 3
```

Comment:
If the `restartCount` field is not present, this key will have a value of `[0]` and an array cannot be compared to a single integer using `Equals`. Please modify this expression so the operator is appropriate. Otherwise, an error will likely be thrown in the logs, which should be avoided.